File: src/subside/plugins/koth/commands/CommandVersion.java
package subside.plugins.koth.commands;

import java.util.ArrayList;
import java.util.List;

import org.bukkit.command.CommandSender;

import subside.plugins.koth.commands.CommandHandler.CommandCategory;
import subside.plugins.koth.utils.MessageBuilder;
import subside.plugins.koth.utils.Perm;

public class CommandVersion extends AbstractCommand {

    public CommandVersion(CommandCategory category) {
        super(category);
    }

    @Override
    public void run(CommandSender sender, String[] args) {
        List<String> list = new ArrayList<>();
        list.add(" ");
        list.addAll(new MessageBuilder("&8========> &2INFO &8<========").buildArray());
        list.addAll(new MessageBuilder("&2Author: &aSubSide").buildArray());

        String version = "&a" + getPlugin().getDescription().getVersion()
                + (getPlugin().getVersionChecker().getNewVersion() != null ? " &7(outdated)" : "");
        list.addAll(new MessageBuilder("&2Version: &a"+ version).buildArray());
        list.addAll(new MessageBuilder("&2Site: &ahttp://bit.ly/1Pyxu2N").buildArray());
        sender.sendMessage(list.toArray(new String[list.size()]));
    }

    @Override
    public Perm getPermission() {
        return Perm.VERSION;
    }

    @Override
    public String[] getCommands() {
        return new String[]{"version"};
    }

    @Override
    public String getUsage() {
        return "/koth version";
    }

    @Override
    public String getDescription() {
        return "Shows the version of this plugin";
    }

}

File: src/subside/plugins/koth/commands/basic/CommandVersion.kt
package subside.plugins.koth.commands.basic

import org.bukkit.command.CommandSender
import subside.plugins.koth.KothManager
import subside.plugins.koth.commands.Command
import subside.plugins.koth.modules.KothHandler
import subside.plugins.koth.utils.IPerm
import subside.plugins.koth.utils.MessageBuilder
import subside.plugins.koth.utils.Perm
import java.util.*

class CommandVersion : Command {
    override fun run(kothManager: KothManager, sender: CommandSender, args: Array<String>) {
        sendVersionInfo(kothManager, sender, Perm.Admin.HELP.has(sender))
    }

    override val permission = Perm.VERSION
    override val commands = arrayOf("version")
    override val usage = "/koth version"
    override val description = "Shows the version of this plugin"


    companion object {
        fun sendVersionInfo(kothManager: KothManager, commandSender: CommandSender, showIfOutdated: Boolean) {
            val list: MutableList<String> = ArrayList()
            list.add(" ")
            list.addAll(MessageBuilder("&8========> &2INFO &8<========").buildArray())
            list.addAll(MessageBuilder("&2Author: &aSubSide").buildArray())

            val version = ("&a" + kothManager.plugin.get()?.description?.version
                    + if (showIfOutdated && kothManager.versionChecker.newVersion != null) " &7(outdated)" else "")
            list.addAll(MessageBuilder("&2Version: &a$version").buildArray())
            list.addAll(MessageBuilder("&2Site: &ahttps://github.com/SubSide/KOTH").buildArray())
            commandSender.sendMessage(list.toTypedArray())
        }
    }
}